adaptived/tests/ftests/ftests.c

```diff
@@ -387,12 +388,109 @@ int stop_transient(const char *transient_name)
 
 #define DEFAULT_CMD	"/bin/bash -c 'while true ; do sleep 1 ; done'"
 #define CG_MNT_PT	"/sys/fs/cgroup"
+#define PROC_MNT	"/proc/self/mounts"
+
+/*
+ * Returns 1 for cgroup v1 and 2 for cgroup v2.
+ * Returns -1 if the version cannot be determined
+ * Does not handle hybrid (both cgroup v1 and v2 on the same machine) cgroups
+ *
+ * NOTE this is not re-entrant
+ */
+int get_cgroup_version(int * const version)
+{
+	bool found_v1 = false, found_v2 = false;
+	struct mntent *ent;
+	FILE *proc_mnt;
+
+	if (!version)
+		return -EINVAL;
+
+	*version = -1;
+
+	proc_mnt = fopen(PROC_MNT, "r");
+	if (!proc_mnt) {
+		adaptived_err("Failed to open %s\n", PROC_MNT);
+		return -EINVAL;
+	}
+
+	ent = getmntent(proc_mnt);
+
+	while (ent) {
+		if (strcmp(ent->mnt_type, "cgroup") == 0)
+			found_v1 = true;
+
+		if (strcmp(ent->mnt_type, "cgroup2") == 0)
+			found_v2 = true;
+
+		ent = getmntent(proc_mnt);
+	}
+
+	if (found_v1)
+		*version = 1;
+
+	if (found_v2 && !found_v1)
+		*version = 2;
+
```

Comment:
Is this needed as well?

```
if (found_v2 && found_v1)
    *version = -1;
```

Reply:
No, that's needed for the v2 case.  On a v1 system, a v2 cgroup (named unified) will also be present, so we need to explicitly check for v2 and v2 only.  Here's what a v1 system shows:
```
$ cat /proc/self/mounts | grep cgroup
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755,inode64 0 0
cgroup2 /sys/fs/cgroup/unified cgroup2 rw,nosuid,nodev,noexec,relatime,nsdelegate 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,name=systemd 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,nosuid,nodev,noexec,relatime,net_cls,net_prio 0 0
cgroup /sys/fs/cgroup/misc cgroup rw,nosuid,nodev,noexec,relatime,misc 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/rdma cgroup rw,nosuid,nodev,noexec,relatime,rdma 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,nosuid,nodev,noexec,relatime,hugetlb 0 0
```

Reply:
I am not familiar with how the hybrid case would be identified. In the hybrid case would both cgroup & cgroup2 mountpoints exist and be mounted?